### DISCUSSION TITLE
Does this library support M5Core2?

### DISCUSSION THREAD

**Discussion by fermintm** [Bodmer/TFT_eSPI#922] [New processor/display support or capability request] [answered]
Does this library support M5Core2?

**Comment by Bodmer** [comment]
No

The M5Core2 Library uses an old version of TFT_eSPI at it;s core so it should be possible to move the relevant setup files over. I am busy at the moment but will look to see if this is possible next week.

**Comment by Bodmer** [comment]
As far as I can tell the ILI9342C is directly compatible with the existing ILI9341 driver, except that it starts up in landscape mode. The library already has the required support. A suitable setup file for the M5Core2 is below. It appears the M5Core2 does not have a backlight control pin. Post back if the backlight LED does not illuminate, or the setup file does not work. I do not have an M5Core2 to test.

```
#define ILI9341_DRIVER

#define TFT_SDA_READ

#define M5STACK             // Adapts ILI9341 driver for the ILI9342C rotation

#define TFT_MISO 38
#define TFT_MOSI 23
#define TFT_SCLK 18
#define TFT_CS   5       // Chip select control pin
#define TFT_DC   15     // Data Command control pin
#define TFT_RST  -1     // Reset pin (could connect to Arduino RESET pin)
#define TFT_BL   -1      // LED back-light (required for M5Stack)

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:-.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

// Comment out the #define below to stop the SPIFFS filing system and smooth font code being loaded
// this will save ~20kbytes of FLASH
#define SMOOTH_FONT

// #define SPI_FREQUENCY   1000000
// #define SPI_FREQUENCY   5000000
// #define SPI_FREQUENCY  10000000
// #define SPI_FREQUENCY  20000000
// #define SPI_FREQUENCY  27000000 // Actually sets it to 26.67MHz = 80/3
#define SPI_FREQUENCY  40000000     // Suggested maximum for ILI9341
// #define SPI_FREQUENCY  80000000

// Reduced SPI frequency for reading TFT
#define SPI_READ_FREQUENCY  16000000
```

**Comment by Bodmer** [comment]
Ah![ Looking here](https://github.com/m5stack/M5Core2) I see the backlight and display power is controlled by a different chip and the TFT_eSPI library does not have a driver for it:
![image](https://user-images.githubusercontent.com/15824805/105784278-675c2500-5f70-11eb-8b24-4f2573a1223f.png)

If you wish to use the M5Core2 I will have to add a driver or a separate function for the sketch.

**Comment by Bodmer** [comment, marked as answer]
There is an [Arduino AXP192 library](https://github.com/tanakamasayuki/I2C_AXP192) that could be used to manage the TFT power. However you would need to look in more detail at what [other functions in the M5Core2 are connected to the chip](https://github.com/m5stack/M5Core2) and whether these need to be controlled. The M5Core2 also has a capacitance touch controller so again you would have to look for a library.

Since the M5Core2 library has an early version of[ TFT_eSPI embedded within](https://github.com/m5stack/M5Core2/blob/master/src/utility/In_eSPI.cpp) it (it has been renamed In_eSPI), the simplest solution is to use that dedicated library.

**Comment by Bodmer** [comment]
@fermintm Thanks for your kind offer of an M5Core2. I have too many projects on the go to make use of one at the moment, so must decline.

**Reply by fermintm** [reply]
😔
